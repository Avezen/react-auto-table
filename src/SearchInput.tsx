import React from 'react';

export interface SearchInputProps {
    search?: string;
    setSearch: any;
}

export const SearchInput = ({search, setSearch}: SearchInputProps) => (
    <input className={'input'} name={'search'} value={search} type={'text'} placeholder={'Search...'} onChange={setSearch} />
);