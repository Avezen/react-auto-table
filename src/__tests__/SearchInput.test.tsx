import React from "react";
import {SearchInput, SearchInputProps} from "../SearchInput";
import renderer from 'react-test-renderer';
import {mount} from "enzyme";

function renderSearchInput(props: Partial<SearchInputProps> = {}) {
    const defaultProps: SearchInputProps = {
        search: 'ok',
        setSearch: () => {}
    };

    return <SearchInput {...defaultProps} {...props} />;
}

describe("<SearchInput />", () => {
    test('should render correctly search component', () => {
        const findByTestId = renderSearchInput(),
            SearchInputComponent = renderer.create(findByTestId).toJSON();
        expect(SearchInputComponent).toMatchSnapshot();
    });

    test('should render search input correctly with empty value', () => {
        const props = {
                search: ''
            };
        const findByTestId = renderSearchInput(props),
            SearchInputComponent = mount(findByTestId).find('input');
        expect(SearchInputComponent.props().value).toEqual('');
    });

    test('should check the onChange callback', () => {
        const setSearch = jest.fn(),

            findByTestId = renderSearchInput({setSearch}),
            SearchInputComponent = mount(findByTestId).find('input');

        SearchInputComponent.simulate('change', { target: { value: 'test' } });
        expect(setSearch).toHaveBeenCalledTimes(1);
    });
});
