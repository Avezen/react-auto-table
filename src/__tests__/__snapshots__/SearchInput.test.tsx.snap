// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> should render correctly search component 1`] = `
<input
  className="input"
  name="search"
  onChange={[Function]}
  placeholder="Search..."
  type="text"
  value="ok"
/>
`;
